application/controllers/QuestionAdministrationController.php

```diff
@@ -2785,50 +2794,73 @@ function ($collector, $oDefaultValue) use ($oQuestion) {
     }
 
     /**
-     * Save subquestion.
+     * Delete sub questions that no longer appear in the post request.
      * Used when survey is *not* activated.
      *
      * @param Question $question
      * @param array $subquestionsArray Data from request.
      * @return void
      * @throws CHttpException
      */
-    private function storeSubquestions($question, $subquestionsArray)
+    private function deleteMissingSubquestions($question, $subquestionsArray)
+    {
+        foreach ($question->subquestions as $subquestion) {
+            if (!isset($subquestionsArray[$subquestion->qid])) {
+                if (!$subquestion->delete()) {
+                    throw (new LSUserException(500, gT("Could not delete subquestion")))
```

Comment:
The really throw error or add error after page display ?

Reply:
LSUserException has frieldy UI if I recall correctly.
Not sure if 500 is the right code.